src/idpi/operators/gis.py

```diff
@@ -0,0 +1,233 @@
+        latitude of the rotated north pole in degrees.
+
+    """
+
+    # all units in degrees
+    rlon: xr.DataArray
+    rlat: xr.DataArray
+    north_pole_lon: float
+    north_pole_lat: float
+
+
+def _check_requirements(geo: Mapping[str, typing.Any]) -> None:
+    requirements = {
+        "angleOfRotationInDegrees": 0.0,
+        "gridType": "rotated_ll",
+        "iScansNegatively": 0,
+        "jScansPositively": 1,
+    }
+    errors = {key: geo[key] for key in requirements if requirements[key] != geo[key]}
+    if errors:
+        msg = f"Unsupported values for keys: {errors}"
+        raise ValueError(msg)
+
+
+def get_grid(geo: Mapping[str, typing.Any]) -> RotLatLonGrid:
+    """Get grid parameters for a given field.
+
+    Only fields defined on regular grids in rotlatlon coordinates,
+    without rotation nor flipped axes are supported.
+
+    Parameters
+    ----------
+    geo : Mapping[str, Any]
+        Grib keys related to the geography of the field.
+
+    Raises
+    ------
+    ValueError
+        if the field does not fulfill the conditions above.
+
+    Returns
+    -------
+    RotLatLonGrid
+        object representing the rotated lat lon grid.
+
+    """
+    _check_requirements(geo)
+
+    ni = geo["Ni"]
+    x0 = geo["longitudeOfFirstGridPointInDegrees"]
+    dx = geo["iDirectionIncrementInDegrees"]
+    rlon = get_grid_coords(ni, x0, dx, "x")
+
+    nj = geo["Nj"]
+    y0 = geo["latitudeOfFirstGridPointInDegrees"]
+    dy = geo["jDirectionIncrementInDegrees"]
+    rlat = get_grid_coords(nj, y0, dy, "y")
+
+    lon_np = (geo["longitudeOfSouthernPoleInDegrees"] - 180) % 360
+    lat_np = -1 * geo["latitudeOfSouthernPoleInDegrees"]
+
+    return RotLatLonGrid(rlon, rlat, lon_np, lat_np)
+
+
+def rot2geolatlon(grid: RotLatLonGrid) -> tuple[xr.DataArray, xr.DataArray]:
+    """Compute geographical lat lon values for a rotated lat lon grid.
+
+    Parameters
+    ----------
+    grid : RotLatLonGrid
+        object representing the rotated lat lon grid.
+
+    Returns
+    -------
+    tuple[xarray.DataArray, xarray.DataArray]
+        tuple of longitudes and latitudes of the grid points in the
+        geographical lat lon CRS.
+
+    """
+    deg2rad = np.pi / 180
+    rad2deg = 180 / np.pi
+    sin_np_lat = np.sin(deg2rad * grid.north_pole_lat)
+    cos_np_lat = np.cos(deg2rad * grid.north_pole_lat)
+
+    sin_np_lon = np.sin(deg2rad * grid.north_pole_lon)
+    cos_np_lon = np.cos(deg2rad * grid.north_pole_lon)
+
+    # Compute new coordinates
+    # ... normalize input coordinates
+    norm_lon = deg2rad * grid.rlon.where(grid.rlon < 180, grid.rlon - 360)
+
+    # ... cache trigonometric operations
+    sin_lat = np.sin(deg2rad * grid.rlat)
+    cos_lat = np.cos(deg2rad * grid.rlat)
+    sin_lon = np.sin(norm_lon)
+    cos_lon = np.cos(norm_lon)
+    # ... compute latitude
+    arg1 = cos_np_lat * cos_lat * cos_lon + sin_np_lat * sin_lat
+    lat = rad2deg * np.arcsin(arg1)
+    # ... compute longitude
+    arg1 = (
```

Comment:
No it's the same way it was done in fieldextra but it does have some benefits in terms of memory footprint

Reply:
I would separate the two here:
- I think in increases readability
- Optimizing memory footprint and shutting the door on potential parallelization seems like premature optimization to me
If we are very certain we want to go the route of minimizing memory footprint because we need to we can go that route though.

Reply:
Can you be specific about the change that you are requesting and what potential parallelisation we are closing the door on?

Reply:
```suggestion
    arg3 = (
```
If we keep naming it arg1 and are writing into the same temporary memory there is a dependency on the lat computation happening before the lon computation which is not necessary here

Reply:
okay but that's not actually how it works in python. The reassignment to arg1 does not reuse the memory space previously allocated by the object that was labelled as arg1. It assigns the arg1 label to a new object in a new memory space. The reference counter for the previously labelled arg1 object then drops to zero and the memory is freed. That's the benefit that I was referring to: the reduction of the peak memory consumption. To enable parallelisation, this operator would need to be split into multiple functions that can be dispatched so the naming of the local variables doesn't really matter. However, I do prefer the immutable style.